src/main/java/in/projecteka/consentmanager/user/PatientsController.java

```diff
@@ -207,8 +210,29 @@ private OtpMediumType getOtpMedium(UserSignUpEnquiry userSignUpEnquiry) {
                                 .switchIfEmpty(Mono.defer(() -> usedTokens.put(caller.getSessionId(), ""))));
     }
 
-    @PostMapping("/recover-cmid")
-    public Mono<RecoverCmIdResponse> recoverCmId(@RequestBody RecoverCmIdRequest request) {
-        return userService.recoverCmId(request);
+    @PostMapping("/profile/recovery-init")
+    public Mono<GenerateOtpResponse> initiateCmIdRecovery(@RequestBody InitiateCmIdRecoveryRequest request) {
+        return isInvalidRecoveryRequest(request)
```

Comment:
Shouldn't it be `isValidRecoveryRequest`?

Reply:
yeah it should be. will change it